source/ranges.tex

```diff
@@ -25,6 +25,19 @@
 
 \indexheader{ranges}%
 \indexlibraryglobal{all_t}%
+\indexlibraryspec{enable_borrowed_range}{subrange}%
+\indexlibraryspec{enable_borrowed_range}{empty_view}%
+\indexlibraryspec{enable_borrowed_range}{iota_view}%
+\indexlibraryspec{enable_borrowed_range}{ref_view}%
+\indexlibraryspec{enable_borrowed_range}{owning_view}%
+\indexlibraryspec{enable_borrowed_range}{take_view}%
+\indexlibraryspec{enable_borrowed_range}{drop_view}%
+\indexlibraryspec{enable_borrowed_range}{drop_while_view}%
+\indexlibraryspec{enable_borrowed_range}{common_view}%
+\indexlibraryspec{enable_borrowed_range}{reverse_view}%
+\indexlibraryspec{enable_borrowed_range}{elements_view}%
+\indexlibraryspec{enable_borrowed_range}{zip_view}%
+\indexlibraryspec{enable_borrowed_range}{adjacent_view}%
```

Comment:
The general approach seems fine, but I'm wondering whether we should instead inline-mark the specializations inside the codeblock, likely together with some text generation, similar to `\libconcept` or `\deflibconcept`. That way, the index entries actually point to the right spot in multi-page synoses.

Reply:
How does my hack look like? I suppose it can be better, but I haven't studied LaTeX.

Reply:
I'm taking it a step back. No hacks.

> before a code block, do it inline before the specialization, assuming that's possible:
> 
> ```
>     @\indexlibraryspec{enable_view}{span}@
>     inline constexpr bool ranges::enable_view<span<ElementType, Extent>> = true;
> ```

I'm leaning toward this for simplicity (extract from https://github.com/cplusplus/draft/pull/5111#discussion_r762068405). That makes index entries point to the right page. Then we can choose how to index harder-to-name specializations. For example, `disable_sized_sentinel_for<reverse_iterator<Iterator1>, reverse_iterator<Iterator2>>` can be unambiguously indexed as `disable_sized_sentinel_for<reverse_iterator>`.

Reply:
Looks good to me.